src/main/java/se/michaelthelin/spotify/model_objects/utils/PlaylistItemFactory.java

```diff
@@ -0,0 +1,22 @@
+package se.michaelthelin.spotify.model_objects.utils;
+
+import com.google.gson.JsonObject;
+import se.michaelthelin.spotify.model_objects.IPlaylistItem;
+import se.michaelthelin.spotify.model_objects.specification.Episode;
+import se.michaelthelin.spotify.model_objects.specification.Track;
+
+public class PlaylistItemFactory {
+  public static IPlaylistItem createPlaylistItem(JsonObject trackObj) {
+    IPlaylistItem item = null;
+    if (trackObj.has("type") && !trackObj.get("type").isJsonNull()) {
```

Comment:
Let's keep the `hasAndNotNull` if we're at it trying to reduce duplicated code, no?

Reply:
Sorry for the delay

Yes, you're right. I fixed this to reuse this function